components/brave_federated_learning/brave_federated_learning_service.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated_learning/brave_federated_learning_service.h"
+
+#include "brave/components/brave_federated_learning/brave_operational_profiling.h"
+#include "brave/components/brave_federated_learning/brave_operational_profiling_features.h"
+#include "brave/components/p3a/pref_names.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+namespace brave {
+
+BraveFederatedLearningService::BraveFederatedLearningService(
+    PrefService* pref_service,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : local_state_(pref_service), url_loader_factory_(url_loader_factory) {}
+
+BraveFederatedLearningService::~BraveFederatedLearningService() {}
+
+void BraveFederatedLearningService::RegisterLocalStatePrefs(
+    PrefRegistrySimple* registry) {
+  BraveOperationalProfiling::RegisterLocalStatePrefs(registry);
+}
+
+void BraveFederatedLearningService::Start() {
+  if (isP3AEnabled() && isOperationalProfilingEnabled()) {
```

Comment:
frankly speaking this class is pretty much useless atm, but we could move the registrar here, to have only upload related logic in `BraveOperationalProfiling`

Reply:
makes sense, moved it 👍